### ISSUE TITLE
Comments (on pastes)

### ISSUE THREAD

**Issue by AKPWebDesign** [jkcclemens/paste#55]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when I paste something and I cannot comment on my paste, or have others comment on it. I think the ability to comment on pastes will be good for soliciting feedback from friends, family, and members of the community.

**Describe the solution you'd like**
I'd like to see some kind of input at the bottom of each paste that allows registered users (and maybe anonymous users?) to comment on the paste.

**Describe alternatives you've considered**
I have considered the alternative of using GitHub Gists, since they do allow comments to be added to Gists, but this is not preferable due to the inability to create an anonymous Gist.

**Additional context**
Here is a screenshot of the GitHub Gist comment box, in relation to files within the Gist:
![GitHub Gist](https://i.imgur.com/OuLoQpF.png)

And here is a mockup of paste.gg with the same functionality:
![paste.gg mockup](https://i.imgur.com/TyJmrDF.png)

**Comment by notriddle** [comment]
Comments are a tarpit of features, bugs, and heuristics; they're very easy to add "because you can", but you end up creating a buggy, undocumented, ad-hoc implementation of half of [Discourse](https://discourse.org/).

Right now, paste.gg is almost useless for cyberbullies and spammers. The latter still hit it, but that's just because they're automated. They're not getting any SEO advantage by spamming it, because the links are all `nofollow`, and they're not getting any human viewers because paste.gg has no list of pastes that exist.

Trolls and bullies have the same problem. Right now, if someone makes a paste that you don't like, or merely someone you don't like makes a paste, *there's nothing you can do about it*. Adding comments gives you an outlet; flood the comments section with crap. Naturally, there's stuff you can do about this; try to remove duplicate (and almost-duplicate) comments, add a way to ban users, rate limit users, browser fingerprinting to recognize users who utilize private-browsing-mode and proxies, special treatment for Tor (ideally a form of special treatment that doesn't involve blocking it outright), moderation tools...

Basically, adding comments turns paste.gg from a mere utility like a wall outlet and turns it into a community platform. Unless you really enjoy babysitting, I recommend against adding comments. They will eat up all your time, and probably hurt your optimistic view of human nature.

**Comment by jkcclemens** [comment]
>..., but you end up creating a buggy, undocumented, ad-hoc implementation of half of Discourse.

Of this I have no doubt. Using a minimal platform for it is an option, as well, but I'd be interested in seeing if that would even be worth it.

>... they're not getting any human viewers because paste.gg has no list of pastes that exist.

That's not entirely true. In the future, there will be the ability to search, at least, and most likely list from the web interface. As it stands, you can view all public pastes via the API (`GET /api/<version>/pastes`). I agree that spammers get very little out of spamming paste instances, though, and we've recently added honeypots which are relatively simple to help fight against the spam in part.

>Trolls and bullies have the same problem. Right now, if someone makes a paste that you don't like, or merely someone you don't like makes a paste, there's nothing you can do about it.

I hadn't really though about this, and I'll be bringing it up both on here and in "team meetings" (aka talking on IRC/Discord). I'm curious about @lol768, @kashike, and @AKPWebDesign's takes on this. While I like the idea of paste spurring collaboration, it seems like the wrong place to do it. paste is for sharing snippets of text, and you need a medium to share a link to a paste, which, presumably, can also be used for discussion.

Taking on comments might be a larger hassle than we really need for no perceivable benefit. To quote @notriddle:

>Basically, adding comments turns paste.gg from a mere utility like a wall outlet and turns it into a community platform.

And I don't think that's within paste's scope.

**Comment by kashike** [comment]
Comments can be very useful on pastes - on GitHub Gist, gists quite often have comments asking for help or giving suggestions to improve, and this is far better than a PM system or emailing.

Allowing paste authors to disable and delete comments would be good to have.

**Comment by notriddle** [comment]
> Comments can be very useful on pastes - on GitHub Gist, gists quite often have comments asking for help or giving suggestions to improve, and this is far better than a PM system or emailing.

Paste.gg probably shouldn't add a PM system, either, for the same reason. It turns paste.gg from a passive document retrieval system into a system that actively *pushes* content to you. GitHub requires an ever-growing system of shadowbanning, rate limiting, Bayesian filtering, a staff of moderators, and a variety of tools for repository owners, and they are still under fire for not doing enough.

> Allowing paste authors to disable and delete comments would be good to have.

That requires you to predict whether a paste is likely to be flooded. It becomes victim blaming when you guess wrong.

Which isn't to say that community platforms shouldn't offer such a feature; they totally should. But offering someone the ability to turn off comments and delete comments after they have already been abused is not an okay answer.

**Comment by jkcclemens** [comment]
It's been almost a month and I've now circled back to this issue again. I don't think comments fall in line with paste's goals. paste is a tool, not a platform.
